hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/HLESurveyTransformationPipelineBuilder.scala

```diff
@@ -146,6 +172,274 @@ object HLESurveyTransformationPipelineBuilder extends PipelineBuilder[Args] {
       ocSecondaryResidenceZip = None,
       ocSecondaryResidenceOwnership = None,
       ocSecondaryResidenceOwnershipOther = None,
-      ocSecondaryResidenceTimePercentage = None
+      ocSecondaryResidenceTimePercentage = None,
+      ddOtherRecentShortTermResidences = Array.empty
+    )
+
+    val withBreed = {
+      val isPureBreed = rawRecord.getRequired("dd_dog_pure_or_mixed") == "1"
+      if (isPureBreed) {
+        val breed = rawRecord.getOptional("dd_dog_breed")
+        dogBase.copy(
+          ddBreedPure = breed,
+          ddBreedPureNonAkc =
+            if (breed.contains("277")) rawRecord.getOptional("dd_dog_breed_non_akc") else None
+        )
+      } else {
+        dogBase.copy(
+          ddBreedMixedPrimary = rawRecord.getOptional("dd_dog_breed_mix_1"),
+          ddBreedMixedSecondary = rawRecord.getOptional("dd_dog_breed_mix_2")
+        )
+      }
+    }
+
+    val withAge = {
+      val ageCertain = rawRecord.getRequired("dd_dog_birth_year_certain") == "1"
+
+      /*
+       * FIXME: How do we encode this in the data?
```

Comment:
I think you did already address this below, when you assign values to ddAgeBasis.

Reply:
Ah, I meant more how we encode the lookup table itself (so people know how to interpret the magic strings). I'll clarify the comment.